src/librustc/ty/query/mod.rs

```diff
@@ -97,228 +98,228 @@ pub use self::on_disk_cache::OnDiskCache;
 // as they will raise an fatal error on query cycles instead.
 define_queries! { <'tcx>
     /// Records the type of every item.
-    [] fn type_of: TypeOfItem(DefId) -> Ty<'tcx>,
+    [] category<Other> fn type_of: TypeOfItem(DefId) -> Ty<'tcx>,
```

Comment:
This is interesting! I like the way you can label the categories so declaratively, but I don't love the syntax. I'd sort of like `#[category(Other)]` or something that looks more like Rust. 

It could also go inside the `[]` -- this was always meant to "house" important flags and so forth. Are we even using that for anything anymore? Maybe we should remove it...

Reply:
Yeah, the current syntax is not very good. My macro-foo is pretty weak so I just did this temporarily. I'd be happy to circle back and try to make it work in the `[]`.